File: FinalProject/Services/ProfileService.cs
namespace FinalProject.Services;

public class ProfileService
{
    private readonly ProfileRepository _profileRepository;
    public ProfileService(ProfileRepository profileRepository)
    {
        _profileRepository = profileRepository;
    }

    internal profile GetProfileById(string profileId)
    {
        profile pro = _profileRepository.GetProfileById(profileId);
        return pro;
    }
}

File: FinalProject.client/src/services/ProfileService.js
import { AppState } from "../AppState.js";
import { logger } from "../utils/Logger.js";
import { api } from "./AxiosService.js";

class ProfileService {

    async getProfileById(profileId) {
        const res = await api.get(`api/profiles/${profileId}`)
        logger.log('here is the profile', res.data)
        AppState.profile = res.data
    }

}
export const profileService = new ProfileService();
